homeassistant/components/zwave_js/climate.py

```diff
@@ -275,6 +288,40 @@ def preset_modes(self) -> Optional[List[str]]:
         """Return a list of available preset modes."""
         return list(self._hvac_presets)
 
+    @property
+    def fan_mode(self) -> Optional[str]:
+        """Return the fan setting."""
+        if (
+            self._fan_mode
+            and self._fan_mode.value is not None
+            and str(self._fan_mode.value) in self._fan_mode.metadata.states
+        ):
+            return cast(str, self._fan_mode.metadata.states[str(self._fan_mode.value)])
+        return None
+
+    @property
+    def fan_modes(self) -> Optional[List[str]]:
+        """Return the list of available fan modes."""
+        if self._fan_mode and self._fan_mode.metadata.states:
+            return list(self._fan_mode.metadata.states.values())
+        return None
+
+    @property
+    def device_state_attributes(self) -> Optional[Dict[str, str]]:
+        """Return the optional state attributes."""
+        if (
+            self._fan_state
+            and self._fan_state.value is not None
+            and str(self._fan_state.value) in self._fan_state.metadata.states
```

Comment:
might need an additional guard for missing metadata here ?

Reply:
nvm, I just remember @MartinHjelmare changed it that an empty list will be returned on the metadata.states

Reply:
Yes, an empty dict by default.